test/Core.Test/AdminConsole/Services/PolicyServiceTests.cs

```diff
@@ -341,6 +360,75 @@ await sutProvider.GetDependency<IPolicyRepository>().Received()
         Assert.True(policy.RevisionDate - utcNow < TimeSpan.FromSeconds(1));
     }
 
+    [Theory, BitAutoData]
+    public async Task SaveAsync_ExistingPolicy_UpdateTwoFactor_WithoutMasterPasswordOr2FA_ThrowsBadRequest(
+        [AdminConsoleFixtures.Policy(PolicyType.TwoFactorAuthentication)] Policy policy, SutProvider<PolicyService> sutProvider)
+    {
+        // If the policy that this is updating isn't enabled then do some work now that the current one is enabled
+
+        var org = new Organization
+        {
+            Id = policy.OrganizationId,
+            UsePolicies = true,
+            Name = "TEST",
+        };
+
+        SetupOrg(sutProvider, policy.OrganizationId, org);
+
+        sutProvider.GetDependency<IPolicyRepository>()
+            .GetByIdAsync(policy.Id)
+            .Returns(new Policy
+            {
+                Id = policy.Id,
+                Type = PolicyType.TwoFactorAuthentication,
+                Enabled = false,
+            });
```

Comment:
It's confusing that this isn't returning the `policy` object from the test parameters.

Either:
* use `Policy policy` in the test parameters and return that from your mock db call, or
* use a `Guid policyId` in the test parameters and create your policy object in the body of the test

Reply:
Same thing here, this was copied from another test. Removed it as it doesn't matter if its new or updating.